tests/test_pygtracker.py

```diff
@@ -18,9 +20,80 @@ def test_version():
 
 # Start tests for generate_course_statistics
 
+
+def test_generate_course_statistics_input_not_exist():
+    tracker = pygtracker.GradeTracker()
+    with raises(ValueError):
+        tracker.generate_course_statistics(course_ids=[500, "511"])
+
+
+def test_generate_course_statistics_input_type1():
+    tracker = pygtracker.GradeTracker()
+    with raises(TypeError):
+        tracker.generate_course_statistics(course_ids=[True, "511"])
+
+
+def test_generate_course_statistics_input_type2():
+    tracker = pygtracker.GradeTracker()
+    with raises(TypeError):
+        tracker.generate_course_statistics(course_ids="511")
+
+
+# the output is equal to the statistics summary calculate mannually
+# def test_generate_course_statistics_equal():
+# 511_mean = num1
+# 522_median = num2
+
+# tracker = pygtracker.GradeTracker()
+# output = tracker.generate_course_statistics()
+
+# assert output[output["course_id"]=="511"][1] == 511_mean
+# assert output[output["course_id"]=="522"][3] == 522_median
+
+# check the columns of the dataframe that come of the function
+
+
+def test_generate_course_statistics_output_match():
+    tracker = pygtracker.GradeTracker()
+    row_num = len(tracker.generate_course_statistics(course_ids=["511", "522"]))
+    columns_list = tracker.generate_course_statistics(course_ids=["511", "522"]).columns
+    assert columns_list[0] == "course_id"
+    assert columns_list[1] == "mean"
+    assert columns_list[2] == "1st-quantile"
+    assert columns_list[3] == "median"
+    assert columns_list[4] == "3rd-quantile"
+    assert row_num == 2
```

Comment:
You haven't tested for the result returned from the function to match what you expected?

Reply:
The test for the expected results are in `test_generate_course_statistics_equal()`, which was commented before